### DISCUSSION TITLE
0 For Each não carrega o conteudo correto

### DISCUSSION THREAD

**Discussion by jcvilanova** [frontendbr/forum#2301] [Discussão]
Pessoal, tenho um Model que faz a leitura através do modo Fullread em uma tabela. Na minha Home chamo o metodo através de um for each para carregar os dados do registro conforme o SELECT que criei. Porem, tenho dois cards, e gostaria que cada eles mostrassem os dados dos ultimos dois registros. Mas...os dois estao mostrando o mesmo registro, tanto o titulo quanto a imagem. Vi em algum lugar que ao invés do for each eu deveria usar um While, pore, não sei como fazer. 

Estou usando o segunte código: 

```php
<?php
    foreach ($this->Dados['sts_artEnocu'] as $artigo) {
        extract($artigo);
?>
```

Ai, faço a exibição conforme o que foi extraido da seguinte forma: 

```php
<div class="col-md-4">
    <div class="card mw-100 shadow-sm rounded-0 card-1" style="background-image: url('<?php echo URL . 'assets/imagens/artigo/' . $imag; ?>'); width: 450px; height: 350px;">
                            
    </div>  
    <div class="card-body">
            <p class="card-text">Este é um card maior e que suporta texto abaixo, </p>
            <div class="d-flex justify-content-between align-items-center">
                    <div class="btn-group">
                            <button type="button" class="btn btn-sm btn-outline-secondary">Ver</button>
                            <button type="button" class="btn btn-sm btn-outline-secondary">Editar</button>
                    </div>
                    <small class="text-muted">9 mins</small>
            </div>
    </div>
                    
</div>

<div class="col-md-4">
    <div class="card mw-100 shadow-sm card-2" style="background-image: url('<?php echo URL . 'assets/imagens/artigo/' . $imag; ?>'); width: 450px; height: 350px;">
                    
    </div>  
    <div class="card-body">
            <p class="card-text">Este é um card maior e que suporta texto abaixo, </p>
            <div class="d-flex justify-content-between align-items-center">
                        <div class="btn-group">
                            <button type="button" class="btn btn-sm btn-outline-secondary">Ver</button>
                            <button type="button" class="btn btn-sm btn-outline-secondary">Editar</button>
```

Mas como falei, os dois cards estão exibindo sempre o mesmo registro e eu gostaria que mostrasse dois registros na sequencia. Se fossem tres cards seria tres registros, 

alguém pode me ajudar ?

**Comment by felquis** [comment]
PHP não é meu ponto forte, mas vamo lá me ajuda a entender também:

- O que é `extract`?
- Da onde vem `$imag`?

Tem como você colocar o trecho do código completo, ou este já é o completo?

PS: se já conseguiu resolver, coloca a solução aqui, obrigado.
